### PR TITLE
Dead Ending Microsoft.CSharp Package and Bumping the leftout assembly versions to 5.0.0.0

### LINKED ISSUE #1918: Assembly version of `Microsoft.CSharp` and `System.Runtime.CompilerServices.Unsafe` in 5.0 is older than the ones in 3.1?
Tested with nightly builds of netcoreapp5.0, this will make app use out of box assembly instead of inbox ones when they are reference from other place.

Seems they are the only two assembly which have older version:

Microsoft.CSharp: 4.0.4 in 5.0, and 4.0.5 in 3.1 and nuget(4.7.0)
System.Runtime.CompilerServices.Unsafe: 4.0.5 in 5.0, and 4.0.6 in 3.1 and nuget(4.7.0)

### PR DESCRIPTION
**Pull request by Anipik**
@333fred @cston @jaredpar just to check if you guys are okay with the no longer producing the Microsoft.CSharp package.

Fixes https://github.com/dotnet/runtime/issues/1918

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]
Looks good from my end (S.R.CS.U)

**Comment by jaredpar** [comment]
> @333fred @cston @jaredpar just to check if you guys are okay with the no longer producing the Microsoft.CSharp package.

What are the implications of doing this? Also do we still produce the Microsoft.VisualBasic package. That is the VB equivalent of Microsoft.CSharp. I would be suspicious if we're in a state where we're producing one but not the other.

**Comment by joperezr** [comment]
> What are the implications of doing this? Also do we still produce the Microsoft.VisualBasic package. That is the VB equivalent of Microsoft.CSharp. I would be suspicious if we're in a state where we're producing one but not the other.

No real implications other than having Microsoft.CSharp's only shipping mechanism being via the shared framework. Usually we try to keep OOBs shipping as package in case we want to add API to it after a release, but with Microsoft.CSharp we can't really do that because it is inbox in many places. The only real way to ship new API via a package for Microsoft.CSharp would be to have it target netcoreapp5.0 (since it targets netstandard2.0 today) and add the API to Netcoreapp 5.0 build only. Because we don't have any plans to do that (and even if we did, we can just do that directly on the shared framework) the best thing is just to stop building the package. M.Vb package was removed about a year ago via this PR: https://github.com/dotnet/corefx/pull/34087

**Review by ViktorHofer — APPROVED** [review]
@ericstj it probably doesn't matter much but should we change the configuration to netcoreapp (latest), away from netstandard2.0?

**Comment by ericstj** [comment]
Agree with @ViktorHofer let's retarget this to netcoreapp5.0 and clean up the [configurations](https://github.com/dotnet/runtime/blob/de5eb348753c688af01cc6aa1a3ee203d0454a01/src/libraries/Microsoft.CSharp/src/Configurations.props#L4-L5).

File: src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il [hunk 1]
```diff
@@ -41,7 +41,7 @@
     01 00 00 00 00
   ) // false
   .hash algorithm 0x00008004
-  .ver 4:0:5:0
+  .ver 5:0:0:0
```
Comment (ericstj) [inline src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il]: Separate issue.  We should see about setting define's to flow the *actual* values from the build-system.  That way we get real fileversion and assembly version in ILProj-built assemblies.

**Review by ericstj — APPROVED** [review]
Please re-target.  While you are at it, you can clean up https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/pkg/descriptions.json#L17-L27

**Comment by joperezr** [comment]
> While you are at it, you can clean up

Can we also clean the BaselineVersion of the package index for Microsoft.CSharp:

https://github.com/dotnet/runtime/blob/2cf4433f64cd10070f19de0c091295108a982b7f/src/libraries/pkg/Microsoft.Private.PackageBaseline/packageIndex.json#L86

**Review by joperezr — APPROVED** [review]
LGTM other than re-targetting and cleanup.

**Comment by jaredpar** [comment]
@joperezr thanks for the explanation. That all makes sense.

What about my questions about Microsoft.VisualBasic though? After reading your explanation I feel more strongly that Microsoft.CSharp and Microsoft.VisualBasic should be consistent here.

**Comment by joperezr** [comment]
I probably shouldn't have used abbreviations for that part of my response 😄 

> M.Vb package was removed about a year ago via this PR: dotnet/corefx#34087

M.Vb I meant Microsoft.VisualBasic, so that package has been removed from our build for one year now. I'm not sure why at that time we didn't decide to also remove Microsoft.CSharp package, since I agree with you that these two should always be somewhat consistent.

**Comment by jaredpar** [comment]
@joperezr 

> M.Vb I meant Microsoft.VisualBasic, so that package has been removed from our build for one year now.

Great! This all seem good to me.

**Review by jaredpar — APPROVED** [review]
Approving the stopping of publishing Microsoft.CSharp as a package. Have not looked specifically at the changes involved here.

**Review by ericstj — APPROVED** [review]

**Comment by ericstj** [comment]
Test failure is due to inbox assembly version change.  Requires temporary suppression here: 
https://github.com/dotnet/runtime/blob/master/src/libraries/pkg/test/frameworkSettings/netcoreapp5.0/settings.targets

Add IgnoredReference for System.Runtime.CompilerServices.Unsafe.

**Comment by jkotas** [comment]
This is causing majority of PRs to fail with:

```
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The located assembly's manifest definition does not match the assembly reference. (0x80131040 (FUSION_E_REF_DEF_MISMATCH))\nFile name: 'System.Runtime.CompilerServices.Unsafe, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'\n at 
```

I am going to submit a revert.

**Comment by ericstj** [comment]
That's interesting seems like a bug in the live-live build.  We should figure out what part of the stack is not running on live bits.

**Pull request merged**